src/codegen/case_translator.cpp

```diff
@@ -58,8 +58,6 @@ codegen::Value CaseTranslator::DeriveValue(CodeGen &codegen,
     }
     when.EndIf(merge_bb);
   }
-  // Jump to the merging block from the internal If merging block
-  codegen->CreateBr(merge_bb);
```

Comment:
After all, the error in the given branch is caused by the codegen.CallPrintf() function calls you added in the DeriveValue().  This is not dummy, but it actually builds and generates a function call.   While doing so, these CallPrintf() code generation seems to complicate the optimization of the case LLVM IR code, which removes a bitcast instruction which should be there.

I recommend to remove the codegen.CallPrintf() call in the constant translator, rather than changing this code.  Meanwhile, I will have a look why this CallPrintf() removes the bitcast operation.